pkg/identity/identity.go

```diff
@@ -21,6 +24,7 @@ type Identity interface {
 // Signer can sign messages using an identity's private credentials.
 type Signer interface {
 	Sign(message []byte) ([]byte, error)
+	Serialize() ([]byte, error)
```

Comment:
This should not be here. A client-supplied signing implementation should not need to know the internal Fabric mechanics of serialising an identity

Reply:
ok. I will see how to modify it.

Reply:
The **Identity** interface already holds all the information we need to internally generate an **msp.SerializedIdentity** protobuf message:

```go
type Identity interface {
	MspID() string       // ID of the Membership Service Provider to which this identity belongs.
	Credentials() []byte // Implementation-specific credentials.
}
```

The code to create and serialize and **msp.SerializedIdentity** is very simple so we can either just do that explicitly in our code when we need the protobuf message, or we can have an _internal_ utility function to do that. There is similar code used in the internal Proposal creation already:

https://github.com/hyperledger/fabric-admin-sdk/blob/ea99b189b606915caabb0c339978757f55336b59/pkg/internal/proposal/transactioncontext.go#L31-L35

We know how to create the correct protobuf message. We should not be requiring the application code to do it for us.